src/hooks/useNavigation.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  ParamListBase,
+  useNavigation as useNavigationNative
+} from '@react-navigation/native'
+import { NativeStackNavigationProp } from '@react-navigation/native-stack'
+
+import { BaseStackParamList } from 'app/components/app-navigator/types'
+
+import { usePushRouteWeb } from './usePushRouteWeb'
+
+type UseNavigationConfig<
+  ParamList extends ParamListBase,
+  RouteName extends keyof ParamList
+> = {
+  native: { screen: RouteName; params: ParamList[RouteName] }
+  web: {
+    route: string
+    fromPage?: string
+    fromNativeNotifications?: string
+  }
+}
+
+type NavigationType = NativeStackNavigationProp<BaseStackParamList>
```

Comment:
`BaseStackParamList` might not be extensive enough for all our stacks, so we might need a type parameter for that at some point. But probably fine for now

Reply:
word, can def try and extend as we use it in more complex places!